### PR TITLE
Build shared object with version number

### PR DESCRIPTION
**Pull request by 1div0**
Please review proposed changes for library building.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -3,10 +3,15 @@
 # Distributed under the OSI-approved BSD 2-Clause License.  See accompanying
 # file `LICENSE` for more details.
 
-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.12.1)
```
Comment (rougier) [inline CMakeLists.txt]: Any reason for this specific version

File: CMakeLists.txt [hunk 2]
```diff
@@ -3,10 +3,15 @@
...
 
 project(freetype-gl LANGUAGES C CXX)
 
+set(LIBRARY_VERSION_MAJOR 1)
+set(LIBRARY_VERSION_MINOR 0)
+set(LIBRARY_VERSION_PATCH 0)
+set(LIBRARY_VERSION_STRING ${LIBRARY_VERSION_MAJOR}.${LIBRARY_VERSION_MINOR}.${LIBRARY_VERSION_PATCH})
```
Comment (rougier) [inline CMakeLists.txt]: You're right we dot not have a library version and we can start with 1.0.0

File: CMakeLists.txt [hunk 3]
```diff
@@ -136,11 +141,14 @@ set(FREETYPE_GL_SRC
     vertex-buffer.c
 )
 
-add_library(freetype-gl STATIC
+add_library(freetype-gl SHARED
```
Comment (rougier) [inline CMakeLists.txt]: Why did you move from static to shared ?

**Comment by rougier** [comment]
Thanks. Travis has failed, can you have look for what is the problem ?

**Comment by 1div0** [comment]
Travis build failed due to old version of CMake. I will try to rebuild with that version and let you know.

CMake Error at CMakeLists.txt:6 (cmake_minimum_required):
  CMake 3.12.1 or higher is required.  You are running version 3.9.2

File: CMakeLists.txt (hunk 2 above)
Comment (adrianbroher) [reply CMakeLists.txt]: Please start with 0.1.0 if you want to introduce a versioning.  `1.0.0` implies `stable`.

Comment (1div0) [reply CMakeLists.txt]: OK, I am struggling with RPM build right now in https://github.com/1div0/freetype-gl

Comment (1div0) [reply CMakeLists.txt]: Just built 64-bit packages as follows:
rpmbuild -ba /usr/src/github.com/1div0/freetype-gl/packaging/rpm/freetype-gl.spec
rpmbuild --target=aarch64 -ba /usr/src/github.com/1div0/freetype-gl/packaging/rpm/freetype-gl.spec

**Comment by rougier** [comment]
Still failing (because of cmake versoin I guess). I'm not familiar enough with CMake to know if a version higher than 3.4.0 is reasonable or not.

**Comment by 1div0** [comment]
For instance, 3.4.0 is used by darktable. However, I am getting following warning:
 CMake Warning (dev) at /usr/share/cmake/Modules/FindOpenGL.cmake:270 (message):
   Policy CMP0072 is not set: FindOpenGL prefers GLVND by default when
   available.  Run "cmake --help-policy CMP0072" for policy details.  Use the
   cmake_policy command to set the policy and suppress this warning.

   FindOpenGL found both a legacy GL library:

     OPENGL_gl_LIBRARY: /usr/lib64/libGL.so

   and GLVND libraries for OpenGL and GLX:

     OPENGL_opengl_LIBRARY: /usr/lib64/libOpenGL.so
     OPENGL_glx_LIBRARY: /usr/lib64/libGLX.so

   OpenGL_GL_PREFERENCE has not been set to "GLVND" or "LEGACY", so for
   compatibility with CMake 3.10 and below the legacy GL library will be used.
 Call Stack (most recent call first):
   CMakeLists.txt:89 (find_package)
 This warning is for project developers.  Use -Wno-dev to suppress it.

So I will try CMake 3.10.

**Comment by yurivict** [comment]
No need to hard-code library type.

PR https://github.com/rougier/freetype-gl/pull/233 is simpler and it allows users to choose the library type.
